builder/job.go

```diff
@@ -171,13 +171,17 @@ func Build(d *daemon.Daemon, buildConfig *Config) error {
 		memory:          buildConfig.Memory,
 		memorySwap:      buildConfig.MemorySwap,
 		cancelled:       buildConfig.WaitCancelled(),
+		imageTag:        tag,
 	}
 
+	defer func() {
+		builder.Daemon.Graph().Release(builder.activeImages, []string{builder.imageTag})
+	}()
```

Comment:
I think it's possible that `builder.imageTag` could be `""`. I'm not sure if this is okay because there may be multiple parallel builds which occur without specifying a tag. If any of them have common `activeImages`, the references may be prematurely released by the first build to finish.

It would probably be safer to just use a random string for the build to hold references to images. There's a util for this: http://godoc.org/github.com/docker/docker/pkg/stringid#GenerateRandomID